src/irmin/irmin.mli

```diff
@@ -1727,18 +1727,26 @@ module Private : sig
           operations. *)
 
       val closure :
-        [> `Read ] t -> min:node list -> max:node list -> node list Lwt.t
+        [> `Read ] t ->
+        min:node list ->
+        max:node list ->
+        (node list, [ `Disconnected of node | `Empty_max ]) result Lwt.t
       (** [closure t ~min ~max] is the transitive closure [c] of [t]'s
           nodes such that:
 
           {ul
-          {- There is a path in [t] from any nodes in [min] to nodes
-          in [c]. If [min] is empty, that condition is always true.}
-          {- There is a path in [t] from any nodes in [c] to nodes in
-          [max]. If [max] is empty, that condition is always false.}
+          {- There is a path in [t] from any nodes in [c] to one node in
```

Comment:
I find this description quite verbose. Perhaps we could have something like:

```
[closure t ~min ~max] is the set of nodes [n] reachable from an element of [max] along a path with no internal vertices in [min].
{ul
  {- If there is a node [n] in [min] that is not reachable from any element of [max], returns [Error `Disconnected n]. }
  {- If [max] is empty, returns [Error `Empty_max]. }
}
```
I'm not sure if this is actually better, but it reads more easily to me :slightly_smiling_face:

Reply:
I rephrased a bit your description as I'm unsure about the internal vertices part. Feel free to change it.

Reply:
I think your rephrasing might introduce an innaccuracy: `Nodes that are reachable from [max] only through paths containing [min] are not included.`. This seems to exclude the elements of `min` from the closure (i.e. the path `max1 -> a -> b -> min1` contains `min1`), which I think is incorrect (but I am not certain). I picked  'interior vertices of the path' to try to avoid this problem.

Reply:
What do you think about `.. along a path which (i) either contains no [min] or (ii) it ends with a [min].` ?

Reply:
Isn't it also valid for the path to start with a `min`? (provided that node is also in `max`)

Reply:
The resulting path will start and end with the same node, meaning that it will not look for any of its ancestors. I think the phrase covers this case. (I'll add a test for this case.)

Reply:
> The resulting path will start and end with the same node

I think not necessarily. This is the example I was thinking of:

![image](https://user-images.githubusercontent.com/26125912/71913518-ddc89200-3177-11ea-949f-45441be6d550.png)

I think your description includes only nodes 3-5 in the closure, since the path 1-3 includes an element of `min`, whereas my expectation of `closure` is that it would include 1-5.

Reply:
It doesn't in this [test](https://github.com/mirage/irmin/pull/908/commits/2687f3ebb45d2b9b3a986f9d456ee606d4106446) which is close to your drawing. In your drawing, node 1 raised error disconnected. 

So the doc is still not clear enough, because indeed node 1 is connected to a max and yet error is returned.

Reply:
That's a shame. Do we have a reason for having the ``Error `Disconnected`` case at all? I definitely wouldn't have expected it here.